stock_rule_buy_check_stock_before/models/procurement_group.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+from odoo.osv import expression
+
+
+class ProcurementGroup(models.Model):
+    _inherit = "procurement.group"
+
+    def _is_product_available(self, product_id, quantity):
+        if quantity == 0:
+            return False
+        return product_id.free_qty >= quantity
+
+    def _get_rule(self, product_id, location_id, values):
+        disable_dropshiping = False
+        sale_line_id = values.get("sale_line_id")
+        if sale_line_id:
+            sale_line = self.env["sale.order.line"].browse(sale_line_id)
+            warehouse = values.get("warehouse_id")
+            if sale_line and self.with_context(
+                warehouse=warehouse.id
+            )._is_product_available(product_id, sale_line.product_uom_qty):
```

Comment:
This computation in `_get_rule` seems to early. I think the approach of mentioned module of Denis is better